### ISSUE TITLE
rowSelectionChanged shouldn't be called with empty selection

### ISSUE THREAD

**Issue by vec2k4** [olifolkerd/tabulator#2622]
Hi Oli,

first let me thank you very much for your work on tabulator, I really appreciate it!

**Describe the bug**
When using options
```javascript
{
  ...
  selectable: true,
  selectableRangeMode: "click",
  rowSelectionChanged: (data, rows) => {
    ...
  },
  ...
}
```
the ```rowSelectionChanged``` callback fires twice if one row is selected or a selection is changed using shift key. First call is with empty ```data``` and ```row``` parameters, second call reflects the current selection.

**Tabulator Info**
Version 4.5.3

**Working Example**
https://jsfiddle.net/vec2k4/bmayfs2z/

**To Reproduce**
1. Go to the fiddle url
2. Open dev tools in browser
3. Select one row
4. See the mentioned calls in console window
5. Extend selection with shift key
6. Again see the mentioned calls in console window

**Expected behavior**
Two suggestions for the expected behavior:
1. Suppress the callback when the selection is empty.
2. Set variable ```selecting``` to true while deselecting rows, so it can be checked in the callback.

See https://github.com/vec2k4/tabulator/commit/74bebae20a893e494ff726bf1b45c64f47a8c0cf for my change implementing the 2nd suggestion.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Chrome
 - Version: 80

**Additional context**
I don't know if this is really an issue or an intended behavior, but implementing the 2nd suggestion shouldn't be harmful and gives us the opportunity to decide if we would like to use the callback while automatically deselecting rows. If requested, I can turn my commit into a pull request.

Best regards

Ralf

**Comment by olifolkerd** [comment]
Hey @vec2k4 

Thanks for the information.

I will look into fixing that for the 4.6 release.

Cheers

Oli :)

**Comment by olifolkerd** [comment]
Hey @vec2k4 

I have pushed a fix for this to the 4.6 branch that will be released later this weekend.

Cheers

Oli :)

**Comment by vec2k4** [comment]
Hey Oli,

works perfecly, thanks again!
